includes/class-sensei-learner.php

```diff
@@ -9,13 +9,107 @@
  * @since 1.9.0
  */
 class Sensei_Learner {
+	/**
+	 * Instance of singleton.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
 	/**
 	 * Cache of the learner terms.
 	 *
 	 * @var WP_Term[]
 	 */
 	private static $learner_terms = [];
 
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Sensei_Course_Enrolment_Manager constructor. Private so it can only be initialized internally.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Sets the actions.
+	 *
+	 * @since 3.0.0
+	 */
+	public function init() {
+		// Delete user activity and enrolment terms when user is deleted.
+		add_action( 'deleted_user', array( $this, 'delete_user_registers' ) );
+	}
+
+	/**
+	 * Delete all user activity and enrolment terms when user is deleted.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @param  integer $user_id User ID.
+	 * @return void
+	 */
+	public function delete_user_registers( $user_id = 0 ) {
+		if ( $user_id ) {
+			// Remove activities.
+			$this->delete_all_user_activity( $user_id );
+
+			// Remove enrolment terms.
+			$learner_term = self::get_learner_term( $user_id );
+			wp_delete_term( $learner_term->term_id, Sensei_PostTypes::LEARNER_TAXONOMY_NAME );
+		}
+	}
+
+	/**
+	 * Delete all activity for specified user.
+	 *
+	 * @since  3.0.0
+	 *
+	 * @param  integer $user_id User ID.
+	 * @return boolean
+	 */
+	public function delete_all_user_activity( $user_id = 0 ) {
+		$dataset_changes = false;
+
+		if ( $user_id ) {
```

Comment:
What about returning `$dataset_changes` if `! $user_id` and avoid a level of indentation?

Reply:
This is great! https://github.com/Automattic/sensei/pull/2898/commits/e0932fe987fdf8500e28dd06110715e041823b07